arch/armv7m/tls.c

```diff
@@ -0,0 +1,34 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libphoenix
+ *
+ * tls.c
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Lukasz Leczkowski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <sys/types.h>
+
+
+volatile addr_t arm_tls_ptr __attribute__((section("armtls"), used)) = 0;
```

Comment:
`addr_t` is reserved for physical addresses, `void *` or `uintptr_t` instead perhaps?